source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1353,17 +1354,24 @@ void RtFileIo::readLibraries(const FilePathVec& libraryPaths, const FileSearchPa
             continue;
         }
 
-        if (elem->isA<Node>())
+        try
         {
-            readNode(elem->asA<Node>(), stage->getRootPrim(), stage, mapper);
-        }
-        else if (elem->isA<NodeGraph>())
-        {
-            readNodeGraph(elem->asA<NodeGraph>(), stage->getRootPrim(), stage, mapper);
+            if (elem->isA<Node>())
+            {
+                readNode(elem->asA<Node>(), stage->getRootPrim(), stage, mapper); 
+            }
+            else if (elem->isA<NodeGraph>())
+            {
+                readNodeGraph(elem->asA<NodeGraph>(), stage->getRootPrim(), stage, mapper);
+            }
+            else
+            {
+                readGenericPrim(elem, stage->getRootPrim(), stage, mapper);
+            }
         }
-        else
+        catch(const ExceptionRuntimeError &e)
         {
-            readGenericPrim(elem, stage->getRootPrim(), stage, mapper);
+            RtApi::get().log(RtLogger::WARNING, e.what());
```

Comment:
Maybe even log as an ERROR since it failed to load.

Reply:
Ok, I guess it's an error still, just not a fatal one. I'll change it.